lib/routes/shared/initial_walkthrough.dart

```diff
@@ -70,31 +71,40 @@ class InitialWalkthroughPageState extends State<InitialWalkthroughPage>
             builder: (_) =>
                 new RestoreDialog(context, widget._backupBloc, options));
       }
+      
+      var restore = (SnapshotInfo snapshot, String key) {
+        widget._backupBloc.restoreRequestSink
+              .add(RestoreRequest(snapshot, key));
+        Navigator.push(
+            context,
+            createLoaderRoute(context,
+                message: "Restoring data...", opacity: 0.8));
+      };
 
-      String restoreKey;
       if (toRestore != null) {
         if (toRestore.encrypted) {
           if (toRestore.encryptionType == "Mnemonics") {
-            restoreKey = await getBackupPhrase();
-            if (restoreKey != null) {
-              await _createBackupPhrase(restoreKey);
-              _updateSecurityModelAction = UpdateSecurityModel(widget._user.securityModel.copyWith(backupKeyType: BackupKeyType.PHRASE));
-            }
-          } else if (toRestore.encryptionType == "Pin") {
-            restoreKey = await getRestorePIN();
-            if (restoreKey != null) _updateSecurityModelAction = UpdateSecurityModel(widget._user.securityModel.copyWith(backupKeyType: BackupKeyType.PIN));
-          }
-          if (restoreKey == null) {
-            _updateSecurityModelAction = UpdateSecurityModel(widget._user.securityModel.copyWith(backupKeyType: BackupKeyType.NONE));
+            restoreUsingPhrase((entrophy) async {
+              await _createBackupPhrase(entrophy);
+              var updateAction = UpdateSecurityModel(widget._user.securityModel.copyWith(backupKeyType: BackupKeyType.PHRASE));
+              widget._registrationBloc.userActionsSink.add(updateAction);
+              restore(toRestore, entrophy);
+            });
             return;
           }
+          
+          if (toRestore.encryptionType == "Pin") {
+            restoreUsingPIN((pin) async {                            
+              var updateAction = UpdateSecurityModel(widget._user.securityModel.copyWith(backupKeyType: BackupKeyType.NONE));
```

Comment:
@roeierez Shouldn't backupKeyType be BackupKeyType.PIN?

Reply:
In case of PIN restore we don't want to update the backup encryption type as this is legacy.